reunion/src/reunion_cfg.cpp

```diff
@@ -36,6 +36,54 @@ bool Reunion_Cfg_LoadDefault()
 	return true;
 }
 
+size_t Reunion_Cfg_GenerateRandomSteamIdSalt(char* steamIdHashSalt)
+{
+	unsigned char iv[16];
+	size_t ret = 0;
+	memset(iv, 0, sizeof(iv));
+#ifdef _WIN32
+	HMODULE m = LoadLibraryA("advapi32");
+	if (m) {
+		void *fn = GetProcAddress(m, "SystemFunction036"); // available in WinXP and later
+		if (fn) {
+			((unsigned char (__stdcall *)(void*,unsigned long))fn)((void *)iv,16);
+		}
+	}
+#else
+	FILE *tmp = fopen("/dev/urandom","rb");
+	if (tmp) {
+		fread((void *)iv,16,1,tmp);
+		fclose(tmp);
+	}
+#endif
+	if (*(uint64_t*)&iv[0] || *(uint64_t*)&iv[8]) {
+		for (unsigned int i = 0; i < 16; i++) {
+			steamIdHashSalt[i*2] = (iv[i] >> 4) < 10 ? (iv[i] >> 4) + 48: (iv[i] >> 4) + 87;
+			steamIdHashSalt[i*2+1] = (iv[i] & 0x0F) < 10 ? (iv[i] & 0x0F) + 48 : (iv[i] & 0x0F) + 87;
+		}
+		steamIdHashSalt[32] = 0;
+		FILE* fl = CReunionConfig::open(REUNION_CFG_FILE, "r+");
+		if (fl) {
+			char linebuf[2048];
+			std::vector<std::string> lines;
+			while (fgets(linebuf, sizeof(linebuf), fl)) {
+				if (!strncmp(linebuf, "SteamIdHashSalt", 15)) {
+					sprintf(linebuf, "SteamIdHashSalt = %s\n", steamIdHashSalt);
```

Comment:
Shut up Codacy, the null termination is right there at line 64 `steamIdHashSalt[32] = 0;`